gomock/matchers.go

```diff
@@ -168,6 +169,33 @@ func (n notMatcher) String() string {
 	return "not(" + n.m.String() + ")"
 }
 
+type regexMatcher struct {
+	regex      *regexp.Regexp
+	compileErr error
+}
+
+func (m regexMatcher) Matches(x any) bool {
+	if m.regex == nil {
+		return false
+	}
+
+	switch t := x.(type) {
+	case string:
+		return m.regex.MatchString(t)
+	case []byte:
+		return m.regex.Match(t)
+	default:
+		return false
+	}
+}
+
+func (m regexMatcher) String() string {
+	if m.compileErr != nil {
+		return m.compileErr.Error()
+	}
+	return "matching regex " + m.regex.String()
```

Comment:
```suggestion
	return "matches regex " + m.regex.String()
```
Rationale: this library has been using something like "has ...", "is ...", or symbols. "matches" seems to match the existing style better.